src/idl_gen_general.cpp

```diff
@@ -650,20 +657,42 @@ class GeneralGenerator : public BaseGenerator {
   // Recursively generate arguments for a constructor, to deal with nested
   // structs.
   void GenStructArgs(const StructDef &struct_def, std::string *code_ptr,
-                     const char *nameprefix) const {
+                     const char *nameprefix, size_t array_count = 0) const {
     std::string &code = *code_ptr;
     for (auto it = struct_def.fields.vec.begin();
          it != struct_def.fields.vec.end(); ++it) {
       auto &field = **it;
-      if (IsStruct(field.value.type)) {
+      auto type = field.value.type;
+      auto array_cnt = array_count;
+
+      if (IsArray(type)) {
+        type = type.VectorType();
+        array_cnt++;
+      }
+
+      if (IsStruct(type)) {
```

Comment:
Maybe, for readability, introduce new local constants?
```cpp
      const auto field_type = field.value.type;
      const auto array_field = IsArray(field_type); 
      const auto type = array_field ? field_type.VectorType() : field_type;
      const auto array_cnt = array_field ? (array_count + 1) : array_count;
```
Please, don't use reassignment if it is possible (`type` variable).

Reply:
Changed.